### PR TITLE
Sad path testing and refactoring 

### PR DESCRIPTION
**Pull request by tylertomlinson**
### Issue(s) related to this PR:
* closes #
#11 #10 #9 #8 #7 #6 #5 #4 #3 #2 #1 

### Type of change made:
* [x] Bugfix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)

#### If this submission is a new feature:
* [x ] Have you added an explanation of what your changes do?
* [x] Have you lint your code locally prior to submission?

### Change Log:
* Ran rubocop on all files 
* Added error handling module 
* Added show action for catches 
* Refactor to DRY code up 
* Added sad path testing 

### Were there any challenges that arose? If so, how were the addressed?
* This options thing

### Technical Debt:
* 

### Testing:
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?
* [x] All new and existing tests passed?
* [x] Have you run linter?

**Test coverage** = 100% models / 100% requests

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -24,4 +24,10 @@ Style/SymbolArray:
   Enabled: false
 
 Metrics/BlockLength:
-  Max: 42
+  Max: 100
+
```
Comment (fredrondina) [inline .rubocop.yml:28]: nice work getting this configured

File: app/controllers/api/v1/catches_controller.rb [hunk 2]
```diff
@@ -25,9 +27,7 @@ def catch_params
     params.permit(:species, :location, :weight, :length, :image, :fly_id)
   end
 
-  # def options
-  #   options = {}
-  #   options[:is_collection] = false
-  #   return options
-  # end
+  def serializer(object)
+    CatchSerializer.new(object)
```
Comment (fredrondina) [inline app/controllers/api/v1/catches_controller.rb:31]: very dry, looking very clean

File: app/controllers/api/v1/flies_controller.rb [hunk 3]
```diff
@@ -30,6 +28,10 @@ def fly_params
     params.permit(:name, :size, :color, :category, :amount, :favorited?)
   end
 
+  def serializer(object)
```
Comment (fredrondina) [inline app/controllers/api/v1/flies_controller.rb:31]: this code is SO DRY

**Review by fredrondina — APPROVED** [review]
Great refactor, code looks super clean and dry. Show path for catches looks great. RuboCop at 0 errors and testing coverage at 100%

File: .rubocop.yml (hunk 1 above)
Comment (tylertomlinson) [reply .rubocop.yml:28]: ;)

**Pull request merged**